### DISCUSSION TITLE
Searchbar mit Filterfunktion (Dropdown)

### DISCUSSION THREAD

**Discussion by anresa** [neuefische/questions#21] [Web] [answered]
Hallo,
ich versuche seit zwei Wochen eine Lösung zu finden und komme nicht weiter. Ich hoffe, ihr könnt mir helfen.

Mein Ziel:
Eine Suchleiste in die ich ein gewünschtes Produkt eingeben kann (von Land A). Danach wähle ich in einem Dropdown Menü ein anderes Land aus (Land B) und dadurch erhalte ich ein equialentes Produkt als Card welche die Suchleiste ersetzt.

Ich habe bisher die Suchleiste gebaut und versucht ein Dropdown Menü anzulegen (als Filter). Außerdem hab ich eine database.js erstellt und diese komplett mit Daten befüllt. Was ich nicht schaffe ist das alles miteinander zu verbinden damit es funktioniert und die richtigen Daten ausspuckt. Ich durchforste seit zwei Wochen das Internet und habe unzählige Tutorials ausprobiert, ich komme alleine nicht weiter!

Meine database.js:

<img width="798" alt="Bildschirmfoto 2022-07-27 um 11 49 28" src="https://user-images.githubusercontent.com/106099037/181218325-1e2fc8b3-367c-4c86-b285-9350a35e5956.png">

Die Suchleiste:

<img width="871" alt="Bildschirmfoto 2022-07-27 um 11 49 52" src="https://user-images.githubusercontent.com/106099037/181218448-60366843-af08-414b-9ed2-013ec0eea10e.png">

Das (unfertige, nicht funktionierende) Dropdown Filter Menü:

<img width="862" alt="Bildschirmfoto 2022-07-27 um 11 50 50" src="https://user-images.githubusercontent.com/106099037/181218577-eb67106a-9001-44f1-91bc-5cf6d5f7d3ba.png">

Daten die ich dafür angelegt haben:

<img width="354" alt="Bildschirmfoto 2022-07-27 um 11 50 11" src="https://user-images.githubusercontent.com/106099037/181218922-080dffc8-1e84-4871-b93d-99d3084bbf0c.png">

Eine der millionen Fehlermeldungen die ich bekomme:

<img width="814" alt="Bildschirmfoto 2022-07-27 um 11 51 02" src="https://user-images.githubusercontent.com/106099037/181219085-5baf51a3-a8a3-401e-8032-2c7823552a51.png">

Mein [veröffentlichtes Repo](https://github.com/anresa/capstone-project) ist aktuell auf dem Stand mit (nur) der Suchleiste da ich seither nur Errors bekomme.

Vielen Dank für eure Hilfe!

**Comment by actyralf** [comment, marked as answer]
Wir haben das Problem persönlich besprochen, das hier wäre die "Architektur", die Du verwenden solltest. **WICHTIG**: Komponenten immer möglichst "dumm" halten. Die Logik (was passiert, wenn der Nutzer den Suchbutton klickt, was, wenn er ein Element in der Liste anklickt etc.) sollte zentral in der App.js verwaltet werden. (Stichwort: State Uplifting)

![mismo-1](https://user-images.githubusercontent.com/99404509/181268217-c268d72b-b044-4ac4-85e1-1dd94c31420d.png)

**Comment by doemser** [comment]
[advanced search filter](https://codesandbox.io/s/advanced-searchbar-filter-czg4uc?file=/src/App.jsl)

here is a small prototype that isolates the challenge
